tests/tasks/generators/clusterloader/load.yaml

```diff
@@ -60,6 +63,10 @@ spec:
       NODE_MODE: master 
       EOL
       cat $(workspaces.source.path)/overrides.yaml
+      cd $(workspaces.source.path)/perf-tests/
+      git fetch origin --verbose --tags
+      git checkout $(params.cl2-branch)
+      git branch
```

Comment:
We should be able to checkout the tag in the earlier step `git-clone`
https://github.com/tektoncd/catalog/blob/main/task/git-clone/0.8/samples/using-git-clone-result.yaml#L36

Reply:
Yeah, moving these commands to that step will make us to move to `script` approach to look clean. As it will be multiple commands which is why i added it here. I don't have hard preference on this.

Reply:
I think we should keep the git cloning and checkout logic in one place